src/v2/guide/custom-directive.md

```diff
@@ -143,37 +143,71 @@ new Vue({
 </script>
 {% endraw %}
 
-Directive arguments can be dynamic. For example, in `v-mydirective:argument=[dataproperty]`, `argument` is the string value assigned to the *arg* property in your directive hook *binding* parameter and `dataproperty` is a reference to a data property on your component instance assigned to the *value* property in the same *binding* parameter. As directive hooks are invoked, the *value* property of the *binding* parameter will dynamically change based on the value of `dataproperty`.
+### Dynamic Directive Arguments
 
-An example of a custom directive using a dynamic argument:
+Directive arguments can be dynamic. For example, in `v-mydirective:[argument]="value"`, the `argument` can be updated based on data properties in our component instance! This makes our custom directives flexible for use throughout our application.
+
+Let's say you want to make a custom directive that allows you to pin elements to your page using fixed positioning. We could create a custom directive where the value updates the vertical positioning in pixels, like this:
 
 ```html
-<div id="app">
+<div id="baseexample">
   <p>Scroll down the page</p>
-  <p v-tack:left="[dynamicleft]">I’ll now be offset from the left instead of the top</p>
+  <p v-pin="200">Stick me 200px from the top of the page</p>
 </div>
 ```
 
 ```js
-Vue.directive('tack', {
-  bind(el, binding, vnode) {
+Vue.directive('pin', {
+  bind: function (el, binding, vnode) {
     el.style.position = 'fixed';
-    const s = (binding.arg == 'left' ? 'left' : 'top');
+    el.style.top = binding.value + 'px';
+  }
+}); 
+
+new Vue({
+  el: '#baseexample'
+});
+```
+
+This would pin the element 200px from the top of the page. But what happens if we run into a scenario when we need to pin the element from the left, instead of the top? Here's where a dynamic argument that can be updated per component instance comes in very handy:
+
+
+```html
+<div id="dynamicexample">
+  <h3>Scroll down inside this section ↓</h3>
+  <p v-pin:[position]="200">I am pinned onto the page at 200px to the left.</p>
+</div>
+```
+
+```js
+Vue.directive('pin', {
+  bind: function (el, binding, vnode) {
+    el.style.position = 'fixed';
```

Comment:
Two last things (I swear 😅):

- I'm afraid a new user may get confused at first by `position` as the dynamic argument's name, since it has a different meaning in CSS (as shown by `el.style.position = 'fixed'`). I couldn't find a better alternative though, so maybe this isn't a big deal after all.
- Did you run the code through a linter which added all these extra semicolons 😄 ? Same with the snippet above.

Reply:
Maybe `side`?

Reply:
- How about `direction`
- I'm a no-semi person, so all the semis are too much for me- we really need to get prettier in here because this is the stuff that doesn't really matter :) I'll run it just the snippet through prettier and change my settings and adjust back to the spaces after parens like we have in the docs. We should make this more automated with a `.prettierrc` in the future.

Reply:
* `direction` sounds good to me
* Totally agreed re. `.prettierrc`, and I asked the question exactly because normally we don't have semi-colons in Vue docs.

Reply:
Oh man, I don't know why I thought we did... working in too many repos at once. I'll remove them, that's easier anyways. Nice catch! Thanks An :)

Reply:
Alright, both should be fixed